### PR TITLE
post: cli 7.1

### PR DESCRIPTION
**Pull request by cexbrayat**
TODO:
- [ ] check `resourcesOutputPath`
- [ ] flag overrides

### CODE REVIEW

File: _posts/2018-11-23-angular-cli-7.1.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+the sizes of your bundles should be slightly reduced (don't expect miracles though).
+
+## @angular/http removed
+
+`@angular/http` has been deprecated for a long time in favor of `@angular/common/http`,
+and it is now removed from the generated project.
+
+## Flag deprecations
+
+The CLI added the possibility to deprecate a flag in a schematics (with the `x-deprecated` field in a schema).
```
Comment (clacote) [inline _posts/2018-11-23-angular-cli-7.1.md]: a (singular) schematics (with an "s")?

File: _posts/2018-11-23-angular-cli-7.1.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+It used to pick the last one silently,
+and now it will still do the same and also warn you that there is something weird going on:
+
+    > ng build --aot --aot=false
+    Option aot was already specified with value true. The new value false will override it.
+
+TODO check in final
+
+That's all for this small release, I hope you enjoyed reading this blog post.
+Stay tuned !
```
Comment (clacote) [inline _posts/2018-11-23-angular-cli-7.1.md]: Stay tuned ! -> Stay tuned! (no space before "!")

File: _posts/2018-11-23-angular-cli-7.1.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+If you want to upgrade to 7.1.0 without pain (or to any other version, by the way), I have created a Github project to help: [angular-cli-diff](https://github.com/cexbrayat/angular-cli-diff). Choose the version you're currently using (6.2.1 for example), and the target version (7.1.0 for example), and it gives you a diff of all files created by the CLI: [angular-cli-diff/compare/6.2.1...7.1.0](https://github.com/cexbrayat/angular-cli-diff/compare/6.2.1...7.1.0).
+It can be a great help along the official `ng update @angular/core @angular/cli` command.
+You have no excuse for staying behind anymore!
+
+Let's see what we've got!
+
+## New resourcesOutputPath option
+
+It is now possible to specify where style resources will be placed, relative to `outputPath`.
+In short you can ouput your CSS in `other` instead of `assets` with:
```
Comment (jnizet) [inline _posts/2018-11-23-angular-cli-7.1.md]: They were never in assets, but were in the root folder, next to index.html.

File: _posts/2018-11-23-angular-cli-7.1.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    "experimentalDecorators": true,
+    "importHelpers": true,
+
+By including these helpers, and avoiding to repeat the same code over and over,
+the sizes of your bundles should be slightly reduced (don't expect miracles though).
+
+## @angular/http removed
+
+`@angular/http` has been deprecated for a long time in favor of `@angular/common/http`,
+and it is now removed from the generated project.
```
Comment (jnizet) [inline _posts/2018-11-23-angular-cli-7.1.md]: from the generated package.json file

File: _posts/2018-11-23-angular-cli-7.1.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+- `skipAppShell` in `ng build/serve` for a Web application as it had no effect.
+
+## Flag override warning
+
+Note that you will now have a warning when you specify a flag several times in the same command.
+It used to pick the last one silently,
+and now it will still do the same and also warn you that there is something weird going on:
+
+    > ng build --aot --aot=false
+    Option aot was already specified with value true. The new value false will override it.
```
Comment (jnizet) [inline _posts/2018-11-23-angular-cli-7.1.md]: will it do something when you specify it twice with the same value, or when you specify it twice due to "meta flags" like 

    ng build --prod --aot 

(that I see a lot in stackoverflow questions)

**Comment by jnizet** [comment]
Sorry, I clicked "Approve", but meant "Request changes". Small changes though.

File: _posts/2018-11-23-angular-cli-7.1.md (hunk 5 above)
Comment (cexbrayat) [reply _posts/2018-11-23-angular-cli-7.1.md]: I was wondering too, but I'm not sure as this has not been in a release yet. I'll give a try when possible and update accordingly.

**Comment by cexbrayat** [comment]
It looks like `resourcesOutputPath` and flag override warning did not make it into the final release. The blog post is even shorter.

File: _posts/2018-11-23-angular-cli-7.1.md (hunk 5 above)
Comment (cexbrayat) [reply _posts/2018-11-23-angular-cli-7.1.md]: So I gave it a try, and no, it doesn't warn you in case of `ng build --prod --aot=false`.

**Pull request merged**